src/askvm/resources/core/assign.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  any,
+  resource,
+  run,
+  runUntyped,
+  Options,
+  TypedValue,
+  JSONable,
+} from '../../lib';
+
+function assignValue(
+  valueObject: { [key: string]: any },
+  key: string,
+  assignedValue: any
+) {
+  if (Object.isFrozen(valueObject[key]))
+    throw new Error(`Cannot assign to a constant variable "${key}"`);
+  valueObject[key] = assignedValue;
+}
+
+export const assignRes = resource({
+  type: any,
+  async compute(options, code, args): Promise<TypedValue<JSONable>> {
+    const { params: children = [] } = code;
+
+    const key: any = await runUntyped(options, children[0]); // FIXME any
+    const value = await run(options, children[1]);
+
+    if (key === 'resources') {
+      throw new Error(`Key "resources" cannot be redeclared`);
+    }
+
+    if (code.name === 'assign') {
```

Comment:
We shouldn't need this. `run` has a wrapper to guarantees that correct resource is called for given resource type

Reply:
Removed

Reply:
I'm having problems with rewriting code executed when this if is true in a way that it assigns value when property key exists and throws when the key doesn't exist so it will take some time 
`

Reply:
Done